lib/run-script.js

```diff
@@ -84,31 +84,52 @@ function runScript (args, cb) {
 
 function list(cb) {
   var json = path.join(npm.localPrefix, "package.json")
+  var cmdList = [ "publish", "install", "uninstall"
+                , "test", "stop", "start", "restart"
+                ].reduce(function (l, p) {
+                  return l.concat(["pre" + p, p, "post" + p])
+                }, [])
   return readJson(json, function(er, d) {
     if (er && er.code !== "ENOENT" && er.code !== "ENOTDIR") return cb(er)
     if (er) d = {}
-    var scripts = Object.keys(d.scripts || {})
+    var allScripts = Object.keys(d.scripts || {})
+    var scripts = [], runScripts = []
+    allScripts.forEach(function (script) {
+      if (cmdList.indexOf(script) !== -1) scripts.push(script)
+      else runScripts.push(script)
+    })
 
     if (log.level === "silent") {
-      return cb(null, scripts)
+      return cb(null, allScripts)
     }
 
     if (npm.config.get("json")) {
       console.log(JSON.stringify(d.scripts || {}, null, 2))
-      return cb(null, scripts)
+      return cb(null, allScripts)
     }
 
-    var s = ":"
-    var prefix = ""
-    if (!npm.config.get("parseable")) {
-      s = "\n    "
-      prefix = "  "
-      console.log("Available scripts in the %s package:", d.name)
+    if (npm.config.get("parseable")) {
+      allScripts.forEach(function(script) {
+        console.log(script + ":" + d.scripts[script])
+      })
+      return cb(null, allScripts)
     }
+
+    var s = "\n    "
+    var prefix = "  "
+    var header = "Available scripts in the %s package"
+    if (scripts) console.log(header + ":", d.name)
     scripts.forEach(function(script) {
       console.log(prefix + script + s + d.scripts[script])
     })
-    return cb(null, scripts)
+    if (!scripts && runScripts) {
```

Comment:
It's weird to have one console.log in a block and the other as a single-statement line. Make both branches into bracketed blocks?